File: business/src/main/java/com/kevinguanchedarias/owgejava/entity/Unit.java
package com.kevinguanchedarias.owgejava.entity;

import com.kevinguanchedarias.owgejava.entity.cache.EntityWithTaggableCache;
import com.kevinguanchedarias.owgejava.entity.listener.EntityWithTaggableCacheListener;
import com.kevinguanchedarias.owgejava.entity.listener.UnitListener;
import lombok.*;
import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.CascadeType;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import jakarta.persistence.*;

import java.io.Serial;
import java.util.List;

@ToString(callSuper = true)
@Entity
@Table(name = "units")
@EntityListeners({
        UnitListener.class,
        EntityWithTaggableCacheListener.class
})
@Data
@EqualsAndHashCode(callSuper = true)
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class Unit extends CommonEntityWithImageStore<Integer> implements EntityWithImprovements<Integer>, EntityWithTaggableCache<Integer> {
    public static final String UNIT_CACHE_TAG = "unit";

    @Serial
    private static final long serialVersionUID = -1923291486680931835L;

    @Column(name = "display_in_requirements")
    private Boolean hasToDisplayInRequirements;

    @Column(name = "order_number")
    private Integer order;

    private Integer points;
    private Integer time;

    @Column(name = "primary_resource")
    private Integer primaryResource;

    @Column(name = "secondary_resource")
    private Integer secondaryResource;

    private Integer energy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "type")
    @Fetch(FetchMode.JOIN)
    private UnitType type;

    private Integer attack;

    @Column(name = "health", nullable = false)
    private Integer health = 1;
    private Integer shield;
    private Integer charge;

    @Column(name = "is_unique", nullable = false)
    private Boolean isUnique = false;

    @Column(name = "can_fast_explore", nullable = false)
    private Boolean canFastExplore = false;

    @Column
    private Double speed = 0D;

    @ManyToOne(fetch = FetchType.LAZY)
    @Fetch(FetchMode.JOIN)
    @JoinColumn(name = "improvement_id")
    @Cascade({CascadeType.MERGE, CascadeType.PERSIST, CascadeType.DELETE})
    private Improvement improvement;

    @Column(name = "cloned_improvements")
    private Boolean clonedImprovements = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "speed_impact_group_id")
    @Fetch(FetchMode.JOIN)
    private SpeedImpactGroup speedImpactGroup;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "attack_rule_id")
    @Fetch(FetchMode.JOIN)
    private AttackRule attackRule;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "critical_attack_id")
    @Fetch(FetchMode.JOIN)
    @Getter
    @Setter
    private CriticalAttack criticalAttack;

    @Column(name = "bypass_shield", nullable = false)
    private Boolean bypassShield = false;

    @Column(name = "is_invisible", nullable = false)
    private Boolean isInvisible = false;

    @OneToMany(mappedBy = "unit", fetch = FetchType.LAZY)
    @ToString.Exclude
    private List<InterceptableSpeedGroup> interceptableSpeedGroups;

    @Column(nullable = false)
    private Integer storedWeight = 1;

    private Long storageCapacity;

    @Override
    public String getCacheTag() {
        return UNIT_CACHE_TAG;
    }
}

File: rust-backend/owge-business/src/model/unit.rs
use serde::{Deserialize, Serialize};

/// Mirrors the `units` table / Java `Unit` entity (the unit catalog).
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Unit {
    pub id: u16,
    pub order_number: Option<u16>,
    pub name: String,
    pub display_in_requirements: Option<i8>,
    pub attack_rule_id: Option<u16>,
    pub image_id: Option<u64>,
    pub points: Option<u32>,
    pub description: Option<String>,
    /// `int` (signed) base build time in seconds.
    pub time: Option<i32>,
    pub primary_resource: Option<u32>,
    pub secondary_resource: Option<u32>,
    pub energy: Option<u16>,
    #[sqlx(rename = "type")]
    pub type_id: Option<u16>,
    pub attack: Option<u16>,
    pub health: Option<u16>,
    pub shield: Option<u16>,
    pub charge: Option<u16>,
    /// `tinyint unsigned`.
    pub is_unique: u8,
    pub can_fast_explore: i8,
    pub speed: Option<f64>,
    pub improvement_id: u16,
    pub cloned_improvements: i8,
    pub speed_impact_group_id: Option<u16>,
    pub critical_attack_id: Option<u16>,
    pub bypass_shield: i8,
    pub is_invisible: i8,
    pub stored_weight: u32,
    pub storage_capacity: Option<u32>,
}
